examples/transactions_example/processor.go

```diff
@@ -0,0 +1,354 @@
+		istate.carsWaiting[istate.currGreen] = 0
+	}
+
+	return true
+}
+
+// FIXME
+// Returns true if output messages were produced, else false.
+func intersectionStateMachine(istate *intersectionState) bool {
+
+	changed := false
+
+	// Elect new green light
+	if time.Since(istate.lastChange) > 4*time.Second {
+		changed = electNewGreenLight(istate)
+	}
+
+	// Get the producer for this istate's input partition
+	producer := producers[istate.partition]
+	if producer == nil {
+		fatal(fmt.Sprintf("BUG: No producer for intersection %s partition %v", istate.name, istate.partition))
+	}
+
+	// Produce message with current intersection light states.
+	isectMsg := intersectionStateMsg{
+		Name:   istate.name,
+		Lights: make(map[string]lightStateMsg)}
+
+	for road := range istate.lightState {
+		if road == istate.currGreen {
+			istate.lightState[road] = "green"
+		} else {
+			istate.lightState[road] = "red"
+		}
+
+		isectMsg.Lights[road] = lightStateMsg{
+			Name:        istate.name,
+			Road:        road,
+			State:       istate.lightState[road],
+			CarsWaiting: istate.carsWaiting[road]}
+	}
+
+	value, err := json.Marshal(isectMsg)
+	if err != nil {
+		fatal(err)
+	}
+
+	err = producer.Produce(
+		&kafka.Message{
+			TopicPartition: kafka.TopicPartition{
+				Topic:     &outputTopic,
+				Partition: kafka.PartitionAny,
+			},
+			Key:   []byte(istate.name),
+			Value: value,
+		}, nil)
+
+	if err != nil {
+		fatal(fmt.Sprintf("Failed to produce message: %v", err))
+	}
+
+	return changed
+}
+
+func trafficLightProcessor(wg *sync.WaitGroup, termChan chan bool) {
+	defer wg.Done()
+
+	doTerm := false
+	ticker := time.NewTicker(500 * time.Millisecond)
+
+	// If punctuate is true the current intersection state is
+	// written to the output topic every ticker interval regardless if
+	// there was a intersection light state change.
+	// Use this to get a more responsive visualization.
+	punctuate := true
+
+	intersectionStates = make(map[string]*intersectionState)
+
+	// The per-partition producers are set up in groupRebalance
+	producers = make(map[int32]*kafka.Producer)
+
+	consumerConfig := &kafka.ConfigMap{
+		"client.id":         "processor",
+		"bootstrap.servers": brokers,
+		"group.id":          processorGroupID,
+		"auto.offset.reset": "earliest",
+		// Consumer used for input to a transactional processor
+		// must have auto-commits disabled since offsets
+		// are committed with the transaction using
+		// SendOffsetsToTransaction.
+		"enable.auto.commit":     false,
+		"go.logs.channel.enable": true,
+		"go.logs.channel":        logsChan,
+	}
+
+	var err error
+	processorConsumer, err = kafka.NewConsumer(consumerConfig)
+	if err != nil {
+		fatal(err)
+	}
+
+	err = processorConsumer.Subscribe(inputTopic, groupRebalance)
+	if err != nil {
+		fatal(err)
+	}
+
+	addLog(fmt.Sprintf("Processor: waiting for messages on topic %s",
+		inputTopic))
+	for !doTerm {
+		select {
+
+		case <-ticker.C:
+			// Run intersection state machine(s) every second.
```

Comment:
every second -> periodically (?). the ticker is 500ms.

though pedantically, periodically is not really right either because of the consumer poll with a timeout.